src/features/form/layoutSets/useCurrentLayoutSetId.ts

```diff
@@ -15,3 +15,24 @@ export function useCurrentLayoutSetId() {
 
   return getLayoutSetIdForApplication({ application, layoutSets, taskId });
 }
+
+export function useCurrentLayoutSet() {
+  const application = useLaxApplicationMetadata();
+  const layoutSets = useLaxLayoutSets();
+  const taskId = useProcessTaskId();
+
+  if (application === ContextNotProvided || layoutSets === ContextNotProvided) {
+    return undefined;
+  }
+
+  const showOnEntry = application.onEntry?.show;
+  if (isStatelessApp(application)) {
+    return layoutSets?.sets.find((set) => set.id === showOnEntry);
+  }
+
+  if (taskId == null) {
+    return undefined;
+  }
+
+  return layoutSets.sets.find((set) => set.tasks?.includes(taskId));
+}
```

Comment:
This may be incorrect and could return a differrent layout set than the above `useCurrentLayoutSetId`